### ISSUE TITLE
allow crossing in grouping variable specification

### ISSUE THREAD

**Issue by bbolker** [lme4/lme4#234]
At the moment one can specify a nested grouping factor on the fly as `~1|foo/bar` (and interactions as `~1|foo:bar`), but `1|foo*bar` doesn't work.  The `findbars()` utility function contains an explicit (and rather complicated) `expandSlash()` formula that expands the former but not the latter.  Is there any reason besides laziness _not_ to try to extend `expandSlash` to handle the crossed case as well?  (i.e., `1|foo*bar` should expand to `(1|foo)+(1|bar)+(1|foo:bar)`).  The only tricky case I can think of is some complex specification with nesting _and_ crossing, `(1|foo/bar*bletch)` -- would that work?  How should order of operations etc. be defined?  Or should we just test for/reject cases with both slashes and stars?

The obvious workaround is to specify the crossing by hand.

```
library(lme4)
dd <- expand.grid(site=factor(1:50),year=factor(1:50),rep=1:10)
## OK
lFormula(~(1|site/year),data=dd,
         control=lmerControl(check.formula.LHS="ignore"))
## fails
lFormula(~(1|site*year),data=dd,
         control=lmerControl(check.formula.LHS="ignore"))
```

**Comment by DavidLansky** [comment]
I don't think this is something you want to pursue. A design with 'crossed random effects' is a very different thing than a design with nested random effects. One of the giant leaps forward in lme4 is its ability to handle designs with crossed and partially crossed random effects. In mer models the difference between a crossed random effects model and a nested random effects model is not the model specification but in the data. It is essential to pay attention to the relationships among the random effects in the data; in particular, to correctly capture a nested design (in mer models) the nested random effect must be 'explicitly nested' in the data frame and not 'implicitly nested.' It is the use of explicit nesting that enables mer functions to sort out (from the data, not from the model specification) which random effects are nested, which are crossed, and which are partially crossed. While crossed error designs are not as common as split-unit designs, they are fairly common. Split-unit designs are often (maybe even typically) analyzed without recognizing the split-unit structure. The rate at which crossed error structures are recognized is far lower than that for split-units. There are plenty of real cases of data with both nested and crossed errors. I think what you are proposing is a different meaning of the term 'crossed' when it is applied to grouping (random) effects than the way it is used in 'a crossed error model.' Perhaps the root of the problem is that crossed fixed effects and crossed random effects are interpreted very differently; we do need think about them differently.

**Comment by dmbates** [comment]
I agree with everything that @DavidLansky said.

The reason for adding the "/" operator was to allow for specification of nested grouping factors using data with an implicit nesting.  In retrospect I'm not sure it was a good idea.  It was included so that those who wanted to reproduce results from a textbook example of from one of the "multilevel modeling" packages could do so without properly recoding the data.  It was not uncommon to have the data for a balanced, nested design displayed as a table as if it were crossed.  I think today's users are more accustomed to proper data management.

I completely agree that introducing a special syntax for crossed grouping factors is unnecessary and a bad idea.
